bin/copy-production-modules.js

```diff
@@ -88,14 +94,17 @@ function processPackage( pkgName, context ) {
 	const subFolderStack = [ pkgFolder, ...folderStack ];
 
 	// iterate its dependencies
-	for ( const depName of pkgDeps ) {
+	for ( const { depName, isOptional } of pkgDeps ) {
 		// Find the dependency in node_modules tree, starting with the package's `node_modules/`
 		// subdirectory and going up the directory tree to the root.
 		const foundFolderIdx = subFolderStack.findIndex( ( folder ) =>
 			fs.existsSync( folder + '/node_modules/' + depName )
 		);
 
-		if ( foundFolderIdx === -1 ) {
+		if ( foundFolderIdx === -1 && isOptional ) {
```

Comment:
Circling back, the real problem is that transitive dependencies may or may not have a "correct" package.json layout for peer/optional deps. We aren't really in control of that, and IMO we're lucky that in this case, we could submit an upstream patch to Gutenberg very easily if we needed to. Since the rest of the build chain and entire app and test suite works fine, does it really make sense to fail the build?

At any rate, I've double checked that the latest version of @wordpress/components (v15) does not include @emotion/native, so once that is upgraded for all dependencies, we should be ok here. Though, I did think the wp monorepo update was blocked until we deprecated IE11 :P Do we need a temporary fix? We could just add react-native to the root package.json temporarily, then it would exist in the directory to be found. We could also create a list of packages we don't care about not existing here.

Reply:
> Circling back, the real problem is that transitive dependencies may or may not have a "correct" package.json layout for peer/optional deps. 

Could you elaborate on that? 🤔 I can see that `emotion/native` does include `react-native` as a `peerDependency`: https://github.com/emotion-js/emotion/blob/main/packages/native/package.json#L29. Or were you referring to another dependency?

Reply:
Ah 💡, I re-read and now I understood. The transitive dependency in question here is `@wordpress/components`, which doesn't include `react-native` as a `peerDependency`, even though it does seem to consume it directly and through other libraries (like `emotion`).

Reply:
I'm a bit confused about considering peer dependencies as optional here, though. Wouldn't that potentially cause other issues at runtime? Isn't what @scinos said [here](https://github.com/Automattic/wp-calypso/pull/53221#issuecomment-889474797) still relevant?

Reply:
In this case, it doesn't cause issues at runtime, because we the react-native portions of @wordpress/components are irrelevant for calypso. We've been using this version of @wordpress/components for many months without issues, and with the missing peer dependency, and it doesn't cause problems elsewhere in the build. It only shows up now because this is the only part of our build chain which crashes if a peer dependency doesn't exist for a transitive dependency, and this is the first time we include @wordpress/components in the server bundle